pkg/admission/mutatingwebhook/plugin.go

```diff
@@ -56,10 +60,16 @@ type Plugin struct {
 	localKubeSharedInformerFactory  kcpkubernetesinformers.SharedInformerFactory
 	globalKubeSharedInformerFactory kcpkubernetesinformers.SharedInformerFactory
 
-	getAPIBindings func(clusterName logicalcluster.Name) ([]*apisv1alpha1.APIBinding, error)
+	getAPIBinding     func(clusterName logicalcluster.Name, gr schema.GroupResource) (*apisv1alpha1.APIBinding, error)
+	getLogicalCluster func(clusterName logicalcluster.Name) (*corev1alpha1.LogicalCluster, error)
 
-	managerLock   sync.Mutex
-	managersCache map[logicalcluster.Name]generic.Source
+	lock  sync.RWMutex
+	cache map[logicalcluster.Name]map[logicalcluster.Name]clusterCache // by request and hook source cluster.
```

Comment:
Can you elaborate why we need sourceCluster and cluster logicalclusters in this cache?

Reply:
Its Source and Target? maybe name them this way if this the case

Reply:
because we have two informers giving us webhook configuration: locally in the workspace and from the APIExport workspace (cache server).